python/setup.py

```diff
@@ -54,7 +55,42 @@ def _copy_backends(active):
         package_data = [f"{os.path.relpath(p, backend_path)}/*" for p, _, _, in os.walk(backend_path)]
         ret.append(
             Backend(name=backend, package_data=package_data, src_dir=curr_path, backend_dir=backend_path,
-                    install_dir=install_dir))
+                    install_dir=install_dir, is_external=False))
+    return ret
+
+def _copy_external_backends():
+    # TRITON_EXTERNAL_CODEGEN_BACKENDS is a semicolon-separated list of backend names
```

Comment:
i'd like to see us have only one environment variable and one CMAKE variable if we can, rather than paired lists.

Could we drop the TRITON_EXTERNAL_CODEGEN_BACKENDS and just use the last element of the path in TRITON_EXTERNAL_CODEGEN_BACKEND_DIRS as the name of the back-end?